services/workflows-service/prisma/schema.prisma

```diff
@@ -304,10 +304,36 @@ model Customer {
   updatedAt DateTime @updatedAt
 
   projects Project[]
+  apiKeys                  ApiKey[]
 
   @@index(name)
   @@index(customerStatus)
-  @@index([authenticationConfiguration(ops: JsonbPathOps)], type: Gin)
+}
+
+enum ApiKeyType {
+  hashed_api_key
+}
+
+model ApiKey {
+  id           String @id @default(cuid())
+
+  customerId    String    
```

Comment:
In the future I think we will be required to link API key with a project, and not with a customer.

Reply:
Lior and I talked about it a bit, and as I told Lior I think project id will be a part of a "scope" an API key will have. but it will always be attached to a customer.